src/services/api.js

```diff
@@ -1,7 +1,84 @@
 import { waterTrackerInstance } from './baseURL';
 
-/* your api routs here */
-export const getContacts = async () => {
-  const { data } = await phonebookInstance.get('/');
-  return data;
+
+// Set and clear token
+const setToken = token => {
+  waterTrackerInstance.defaults.headers.common.Authorization = `Bearer ${token}`;
+};
+
+const clearToken = () => {
+  waterTrackerInstance.defaults.headers.common.Authorization = '';
 };
+
+// User apis
+export const getCurrentUser = async (userId) => {
+  const {data} = await waterTrackerInstance.get(`/users/current`);
+};
+
+export const userLogin = async (userData) => {
+  const {data} =  await waterTrackerInstance.post('/users/login', userData);
+  setToken(data.token);
+  return data
+
+}
+
+export const userRegister = async(userData) => {
+  const {data} = await waterTrackerInstance.post('/users/register', userData)
+  setToken(data.token);
+  localStorage.clear();
+  return data
+}
+
+export const userLogOut = async (userData) => {
+  const {data} = await waterTrackerInstance.post('/users/logout', userData);
+  clearToken()
+}
+
+export const uploadUserAvatar = async(userData) => {
+  const {data} = await waterTrackerInstance.patch('/users/udateAvatar', userData)
+  return data
+}
+
+export const updateUser = async(userData) => {
+  const {data} = await waterTrackerInstance.patch('/users/updateUser', userData)
+}
```

Comment:
api's function should return anything . Change {data} on {result} and return results